### ISSUE TITLE
Consolidate closing/removing approach across components

### ISSUE THREAD

**Issue by jcfranco** [Esri/calcite-design-system#6789]
### Description

This stems from https://github.com/Esri/calcite-components/pull/6775#discussion_r1167281298:

> We should consolidate how we handle closed/removed items. I've noticed we have the following discrepancies:
> 
> 1. naming: `closable` vs `removable`
> 2. approach: hiding vs removing the node

cc @driskull 

### Proposed Advantages

Consistency.

### Which Component

All closables/removables:

| component | prop name | approach | 
|--------|--------|--------|
| alert | closable | hidden | 
| chip | closable | hidden | 
| flow-item | closable | hidden | 
| list-item | closable | hidden |  
| ~modal~ (deprecated) | closable | hidden | 
| notice | closable | hidden | 
| panel | closable | hidden | 
| ~pick-list-item~ (deprecated) | ~closable~ | ~removed~ | 
| popover | closable | hidden | 
| ~tip~ (deprecated) | ~closedDisabled~ | ~hidden~ | 
| ~value-list-item~ (deprecated) | ~removable~ | ~removed~ |

**Comment by driskull** [comment]
I think the only "removable" ones are `flow-item` and `value-list-item`. We can possibly ignore `value-list-item` as we plan on deprecating it. `flow-item` is pretty unique and we did have an issue to change it to not remove any child elements but we kind of scrapped that for now. We could revisit it but it would be a breaking change.

**Comment by driskull** [comment]
A lot of the nodes where we set them to `hidden` instead of removing them is because inside of those nodes we have a slot element and we don't want to remove the slot element conditionally in order to keep using `onslotchange` events.

**Comment by geospatialem** [comment]
Blocked by the efforts of https://github.com/Esri/calcite-design-system/issues/6473.

**Comment by eriklharper** [comment]
I'm just now learning about certain parent <> child component relationships where slotted children are physically removed from the DOM, and my first instinct is that this is an anti-pattern because the parent doesn't own the slotted children, the HTML author does, so shouldn't it be up to the HTML author to perform the node removal rather than the calcite component?  I'm thinking of VDOM use cases too, where there is an expectation that the DOM reflects the app's internal state.

**Comment by driskull** [comment]
Yes, its an anti pattern. `flow-item` is the only one doing this and it will be fixed in 3x with a breaking change. At 3x, we can close this issue. https://github.com/Esri/calcite-design-system/pull/9390 https://github.com/Esri/calcite-design-system/issues/5072

panel doesn't remove elements so this chart should be updated.

I think we can close this issue and rely on #5072 cc @geospatialem

**Comment by jcfranco** [comment]
This was addressed by https://github.com/Esri/calcite-design-system/issues/5072.
